molo/surveys/models.py

```diff
@@ -428,6 +432,17 @@ def serve(self, request, *args, **kwargs):
             form = self.get_form(request.POST, page=self, user=request.user)
```

Comment:
thinking all this `if request.method == 'POST':` logic block can be encapsulated within a "handle Post method" to keep the code legible